pkg/acceptance/barecluster/localcluster.go

```diff
@@ -116,16 +142,39 @@ func (c *Cluster) Start(
 	c.rpcCtx = rpc.NewContext(log.AmbientContext{Tracer: tracing.NewTracer()}, baseCtx,
 		hlc.NewClock(hlc.UnixNano, 0), c.stopper)
 
-	if perNodeArgs != nil && len(perNodeArgs) != len(c.Nodes) {
-		panic(fmt.Sprintf("there are %d nodes, but perNodeArgs' length is %d",
-			len(c.Nodes), len(perNodeArgs)))
+	var wg sync.WaitGroup
+	for i := range c.Nodes {
+		cfg := c.cfg.PerNodeCfg[i] // zero value is ok
+		if cfg.DataDir == "" {
+			cfg.DataDir = filepath.Join(c.cfg.DataDir, fmt.Sprintf("%d", i+1))
+		}
+		if cfg.Addr == "" {
+			cfg.Addr = "127.0.0.1"
+		}
+		if cfg.DB == "" {
+			cfg.DB = c.cfg.DB
+		}
+		if cfg.NumWorkers == 0 {
+			cfg.NumWorkers = c.cfg.NumWorkers
+		}
+		cfg.ExtraArgs = append(append([]string(nil), c.cfg.AllNodeArgs...), cfg.ExtraArgs...)
+		c.Nodes[i] = c.makeNode(&wg, i, c.cfg.Binary, cfg)
+		if i == 0 && cfg.RPCPort == 0 {
+			// The first node must know its RPCPort or we can't possibly tell
+			// the other nodes the correct one to go to.
+			//
+			// TODO(tschottdorf): this limits being able to set up a cluster
+			// first and clone it for each test, because all ports change so the
+			// cluster won't come together. Luckily, it takes only ~2 seconds
+			// from zero to a replicated 4 node cluster.
```

Comment:
Is there anything actionable about this `TODO`? I'm not seeing a way to avoid having to know the RPCPort of the first node. Perhaps `s/TODO(tschottdorf)/Note:/g`.